iceoryx2-pal/posix/src/macos/mman.rs

```diff
@@ -74,23 +78,94 @@ unsafe fn shm_file_path(name: *const char, suffix: &[u8]) -> [u8; MAX_PATH_LENGT
     state_file_path
 }
 
+unsafe fn get_real_shm_name(name: *const char) -> Option<[u8; SHM_MAX_NAME_LEN]> {
+    let shm_file_path = shm_file_path(name, SHM_STATE_SUFFIX);
+    let shm_state_fd = open_with_mode(shm_file_path.as_ptr().cast(), O_RDONLY, 0);
```

Comment:
Just to get it right. It's possible to creates files with more than 33 characters in /dev/shm, e.g. text files but it is not possible to create a shared memory file with more than 33 characters, right?

Reply:
Correct, but the limit is 31. The buffer here has a size of 33 since this is the theoretical maximum that will be never reached.

Here the link to the shm name length: https://stackoverflow.com/questions/38049068/osx-shm-open-returns-enametoolong

Reply:
Oh, then in theory 3 `u32` + 2 `_` can result in 32 characters. Practically this will never happen since you use only 9 characters for the nanoseconds ... and the PID and seconds part should also not become that large.

Reply:
Pid and seconds can be at most 2^32 which is equivalent to 10 ascii characters + 9 ascii characters for nanoseconds + 2 `_`. So 31 characters exactly.